src/regex.cr

```diff
@@ -112,6 +112,20 @@ require "./regex/*"
 # groups can also be given names, using the `(?&lt;name&gt;...)` syntax, as in the
 # previous example.
 #
+# Following a match, the special variables $N (e.g., $1, $2, $3, ...) can be used
+# to access a capture group. Trying to access an invalid capture group will raise an
+# exception. Note that it is possible to have a successful match with a nil capture:
+#
+# ```
+# /(spice)(s)?/.match("spice") # => #<Regex::MatchData "spice" 1:"spice" 2:nil>
+# $1                           # => "spice"
+# $2                           # => Exception
```

Comment:
The best comment for exception will be `# raises Exception`, following the other inline examples (see #3820)